src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -43,10 +165,10 @@ class DirectStepI : public StepImp<IOUAmount, IOUAmount, DirectStepI>
     // Charge transfer fees when the prev step redeems
     Step const* const prevStep_ = nullptr;
 
-    // When offerCrossing then ignore trust line quality.
-    bool const offerCrossing_;
+    // Policy that distinguishes behavior between payments and offer crossing.
+    Policy policy_;
+    friend Policy;
```

Comment:
Consider the direct step policy classes. There are five functions defined there. Three of them pass in the `step` as a parameter and these policy classes are friends. It seems cleaner to implement that kind of relationship through inheritance.

This kind of relationship is also showing up in new code. In the optimization commit. `qualityUpperBound` passes in three parameters that are member vars on the step class. It seems cleaner if `qualityUpperBound` was a member of the step class rather than passing that state in.